### PR TITLE
[BIOMAGE-1002] Remove delay from Project Name validation 

### PR DESCRIPTION
**Pull request by xverges**
# Background
#### Link to issue 
https://biomage.atlassian.net/browse/BIOMAGE-1002
#### Link to staging deployment URL 
https://ui-xavier-ui282-create-proj.scp-staging.biomage.net
#### Anything else the reviewers should know about the changes here

# Changes
### Code changes
Validation now happens inside a `useEffect` hook. Does not solve https://biomage.atlassian.net/browse/BIOMAGE-1018

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [ ] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: src/components/data-management/NewProjectModal.jsx [hunk 1]
```diff
@@ -44,6 +35,27 @@ const NewProjectModal = (props) => {
...
+
+  const {
+    saving,
+    error,
+  } = useSelector((state) => state.projects.meta);
+
+  const submit = () => {
+    // TO-DO: this fails to clear the input field
+    // We prpbably can fix this taking advantage of
+    // the Form properties.
```
Comment (aerlaut) [inline src/components/data-management/NewProjectModal.jsx]: Is this solved with `onFinish='submit'` on line 95?

**Review by aerlaut — COMMENTED** [review]
Small comment, otherwise LGTM. Can you put up a staging link too?

File: src/components/data-management/NewProjectModal.jsx (hunk 1 above)
Comment (xverges) [reply src/components/data-management/NewProjectModal.jsx]: No, that's a failed attempt. I am going to remove it now

File: src/components/data-management/NewProjectModal.jsx [hunk 2]
```diff
@@ -49,7 +49,7 @@ const NewProjectModal = (props) => {
 
   const submit = () => {
     // TO-DO: this fails to clear the input field
-    // We prpbably can fix this taking advantage of
+    // We probably can fix this taking advantage of
     // the Form properties.
```
Comment (aerlaut) [inline src/components/data-management/NewProjectModal.jsx]: Let's not leave TO-DOs inside the code. Better turn todos into a task in Jira.

**Review by aerlaut — CHANGES_REQUESTED** [review]
LGTM after removing the TO-DO comment.

File: src/components/data-management/NewProjectModal.jsx (hunk 2 above)
Comment (xverges) [reply src/components/data-management/NewProjectModal.jsx]: I think I like better having both of them 🤔 
Upaded https://biomage.atlassian.net/browse/BIOMAGE-1018 with the comment above and removed it in the code

**Review by aerlaut — APPROVED** [review]
LGTM, tested in staging

**Pull request merged**